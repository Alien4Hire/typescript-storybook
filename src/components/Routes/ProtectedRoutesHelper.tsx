import React from 'react';
import { Navigate, useLocation } from 'react-router-dom';
import { useAppSelector } from '../../app/hooks';
import { loginPathName } from '../Pages/LoginPage';
import { selectUser } from '../../app/reducers/login/loginSlice';

type ProtectedRoutePropsT = {
  children: JSX.Element;
};

const ProtectedRoutesHelper = ({ children }: ProtectedRoutePropsT) => {
  const user = useAppSelector(selectUser);
  const location = useLocation();

  if (!user) {
    const navState = { location };
    return <Navigate to={loginPathName} replace={true} state={navState} />;
  }

  return children;
};

export default ProtectedRoutesHelper;
