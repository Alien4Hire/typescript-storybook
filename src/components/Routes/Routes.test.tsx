import { Suspense } from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { render, screen, act } from '@testing-library/react';
import { ThemeProvider, StyledEngineProvider } from '@mui/material/styles';

import { Provider } from 'react-redux';
import { store } from '../../app/store';

import ProtectedRoutesHelper from './ProtectedRoutesHelper';
import Routes from './Routes';

import { theme } from '../../theme';
import '../../setupTests';

jest.mock('./ProtectedRoutesHelper');
jest.mock('../../app/reducers/config/configApi', () => {
  return Promise.resolve({
    googleClientKey: 'mock-client-key',
  });
});

const folderPageTestId = 'FolderPage';
const activitySettingsPageTestId = 'ActivitySettingsPage';
const searchPageTestId = 'SearchPage';
const templatesPageTestId = 'TemplatesPage';
const templatesEditPageTestId = 'TemplatesEdit';
const templatesNewVersionPageTestId = 'TemplatesNewVersion';
const suspenseTestId = 'SuspenseTestId';

jest.mock('../Pages/FolderPage', () => () => (
  <div data-testid={folderPageTestId} />
));
jest.mock('../Pages/ActivitySettingsPage', () => () => (
  <div data-testid={activitySettingsPageTestId} />
));
jest.mock('../Pages/SearchPage', () => () => (
  <div data-testid={searchPageTestId} />
));
jest.mock('../Pages/TemplatesPage', () => () => (
  <div data-testid={templatesPageTestId} />
));
jest.mock('../Pages/TemplatesPage/TemplateEdit', () => () => (
  <div data-testid={templatesEditPageTestId} />
));
jest.mock('../Pages/TemplatesPage/TemplateNewVersion', () => () => (
  <div data-testid={templatesNewVersionPageTestId} />
));

const Component = ({ pathname = '/' } = {}) => (
  <Provider store={store}>
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={theme}>
        <Suspense fallback={<div data-testid={suspenseTestId} />}>
          <Router initialEntries={[pathname]}>
            <Routes />
          </Router>
        </Suspense>
      </ThemeProvider>
    </StyledEngineProvider>
  </Provider>
);

const renderComponent = ({ pathname = '/' }) => {
  return render(<Component pathname={pathname} />);
};

describe('<Routes /> Component', () => {
  const mockedProtectedRoutesHelper = ProtectedRoutesHelper as jest.Mock;

  beforeEach(() => {
    mockedProtectedRoutesHelper.mockImplementation(({ children }) => {
      return children;
    });
  });

  afterEach(() => {
    mockedProtectedRoutesHelper.mockClear();
  });

  it('renders without crashing', () => {
    const view = renderComponent({});
    expect(view).toMatchSnapshot();
  });

  describe('When a route is protected and the user is logged in', () => {
    it('should show suspense fallback', () => {
      renderComponent({
        pathname: '/templates/:id/activities/:templateVersionId',
      });
      expect(screen.getByTestId(suspenseTestId)).toBeInTheDocument();
    });

    it('should render edit template metadata page', async () => {
      renderComponent({ pathname: '/templates/:id/edit/:templateVersionId' });
      await act(async () => {
        expect(
          await screen.findByTestId(templatesEditPageTestId)
        ).toBeInTheDocument();
      });
    });

    it('should render new version form template metadata page', async () => {
      renderComponent({
        pathname: '/templates/:templateId/version/:templateVersionId',
      });
      await act(async () => {
        expect(
          await screen.findByTestId(templatesNewVersionPageTestId)
        ).toBeInTheDocument();
      });
    });

    it('should render folder page', async () => {
      renderComponent({
        pathname: '/templates/:id/activities/:templateVersionId',
      });
      await act(async () => {
        expect(await screen.findByTestId(folderPageTestId)).toBeInTheDocument();
      });
    });

    it('should render activity settings page', async () => {
      renderComponent({
        pathname: '/templates/:id/activity-settings/:templateVersionId',
      });
      await act(async () => {
        expect(
          await screen.findByTestId(activitySettingsPageTestId)
        ).toBeInTheDocument();
      });
    });

    it('should render search page', async () => {
      renderComponent({ pathname: '/templates/:id/search/:templateVersionId' });
      await act(async () => {
        expect(await screen.findByTestId(searchPageTestId)).toBeInTheDocument();
      });
    });

    it('should render templates page', async () => {
      renderComponent({ pathname: '/templates' });
      await act(async () => {
        expect(
          await screen.findByTestId(templatesPageTestId)
        ).toBeInTheDocument();
      });
    });

    it('should redirect to templates page', async () => {
      renderComponent({ pathname: '/' });
      await act(async () => {
        expect(
          await screen.findByTestId(templatesPageTestId)
        ).toBeInTheDocument();
      });
    });
  });

  describe('When a route is protected and the user has not logged in', () => {
    it('should redirect the user to the Login screen', () => {
      mockedProtectedRoutesHelper.mockClear();
      renderComponent({
        pathname: '/templates/:id/activities/:templateVersionId',
      });
      const loginNode = screen.getByText(/Login/i);
      expect(loginNode).toBeInTheDocument();
    });
  });
});
