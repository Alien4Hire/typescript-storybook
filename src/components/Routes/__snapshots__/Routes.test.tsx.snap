// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> Component renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  background-color: #ffffff;
  color: rgba(0,0,0,0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

.c6 {
  margin: 0;
  color: #080808;
  font-size: 18px;
  font-family: Source Sans Pro;
  font-weight: 600;
  line-height: 1.167;
}

.c7 {
  margin: 0;
  color: #080808;
  font-size: 16px;
  font-family: Source Sans Pro;
  font-weight: 600;
  line-height: 1.235;
  color: #686868;
}

.c17 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: 1.5rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  box-sizing: border-box;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: fixed;
  z-index: 1100;
  top: 0;
  left: auto;
  right: 0;
  background-color: #00758e;
  color: #fff;
  background-color: #142A35;
}

.c9 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.4375em;
  color: rgba(0,0,0,0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9.Mui-disabled {
  color: rgba(0,0,0,0.38);
  cursor: default;
}

.c14 {
  font: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
}

.c14::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c14::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c14:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c14::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c14:focus {
  outline: 0;
}

.c14:invalid {
  box-shadow: none;
}

.c14::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false] + .MuiInputBase-formControl .c13::-webkit-input-placeholder {
  opacity: 0 !important;
}

label[data-shrink=false] + .MuiInputBase-formControl .c13::-moz-placeholder {
  opacity: 0 !important;
}

label[data-shrink=false] + .MuiInputBase-formControl .c13:-ms-input-placeholder {
  opacity: 0 !important;
}

label[data-shrink=false] + .MuiInputBase-formControl .c13::-ms-input-placeholder {
  opacity: 0 !important;
}

label[data-shrink=false] + .MuiInputBase-formControl .c13:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false] + .MuiInputBase-formControl .c13:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false] + .MuiInputBase-formControl .c13:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false] + .MuiInputBase-formControl .c13:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.c14.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0,0,0,0.38);
}

.c14:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.c19 {
  text-align: left;
  position: absolute;
  bottom: 0;
  right: 0;
  top: -5px;
  left: 0;
  margin: 0;
  padding: 0 8px;
  pointer-events: none;
  border-radius: inherit;
  border-style: solid;
  border-width: 1px;
  overflow: hidden;
  min-width: 0%;
}

.c21 {
  float: unset;
  overflow: hidden;
  padding: 0;
  line-height: 11px;
  -webkit-transition: width 150ms cubic-bezier(0.0,0,0.2,1) 0ms;
  transition: width 150ms cubic-bezier(0.0,0,0.2,1) 0ms;
}

.c10 {
  position: relative;
  border-radius: 4px;
}

.c10:hover .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0,0,0,0.87);
}

.c10.Mui-focused .MuiOutlinedInput-notchedOutline {
  border-color: #00758e;
  border-width: 2px;
}

.c10.Mui-error .MuiOutlinedInput-notchedOutline {
  border-color: #d32f2f;
}

.c10.Mui-disabled .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0,0,0,0.26);
}

.c20 {
  border-color: rgba(0,0,0,0.23);
}

.c15 {
  padding: 16.5px 14px;
}

.c15:-webkit-autofill {
  border-radius: inherit;
}

.c12 {
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-radius: 4px;
  cursor: pointer;
}

.c12:focus {
  border-radius: 4px;
}

.c12::-ms-expand {
  display: none;
}

.c12.Mui-disabled {
  cursor: default;
}

.c12[multiple] {
  height: auto;
}

.c12:not([multiple]) option,
.c12:not([multiple]) optgroup {
  background-color: #ffffff;
}

.c12.c12.c12 {
  padding-right: 32px;
}

.c12.MuiSelect-select {
  height: auto;
  min-height: 1.4375em;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c18 {
  position: absolute;
  right: 7px;
  top: calc(50% - .5em);
  pointer-events: none;
  color: rgba(0,0,0,0.54);
}

.c18.Mui-disabled {
  color: rgba(0,0,0,0.26);
}

.c16 {
  bottom: 0;
  left: 0;
  position: absolute;
  opacity: 0;
  pointer-events: none;
  width: 100%;
  box-sizing: border-box;
}

.c2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 16px;
  padding-right: 16px;
  min-height: 56px;
}

.c3 {
  padding: 0px;
}

.c4 {
  margin-top: 5px;
  height: 45px;
  width: 45px;
}

.c8 {
  margin-left: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c11 {
  width: 90px !important;
  height: 60px !important;
  margin: 0 24px;
  color: #ffffff;
  margin-left: 24px;
  position: relative;
}

.c11 .MuiSelect-select {
  padding: 0 13px 0 0 !important;
}

.c11 fieldset {
  border: none;
}

.c11 .MuiSelect-icon {
  fill: #ffffff;
  right: 0;
  width: 20px;
  height: 20px;
}

.c22 {
  margin: 64px 0 0 0;
}

@media print {

}

@media (hover:none) {

}

@media print {
  .c1 {
    position: absolute;
  }
}

@media (hover:none) {
  .c10:hover .MuiOutlinedInput-notchedOutline {
    border-color: rgba(0,0,0,0.23);
  }
}

@media (min-width:660px) {
  .c2 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:0px) and (orientation:landscape) {
  .c2 {
    min-height: 48px;
  }
}

@media (min-width:660px) {
  .c2 {
    min-height: 64px;
  }
}

<body>
    <div>
      <div
        style="display: none;"
      >
        <header
          class="c0 MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 c1 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed"
        >
          <div
            class="c2 MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular c3"
          >
            <a
              href="/"
              style="padding: 5px 10px 5px 0px;"
            >
              <svg
                class="c4"
              >
                macmillan_flag.svg
              </svg>
            </a>
            <div
              class="c5"
            >
              <h3
                class="c6 MuiTypography-root MuiTypography-h3"
              />
              <h4
                class="c7 MuiTypography-root MuiTypography-h4"
              />
            </div>
            <div
              class="c8"
            >
              <div
                class="c9 c10 MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiSelect-root-kTniIo c11"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  class="c12 MuiSelect-select MuiSelect-outlined c13 c14 c15 MuiOutlinedInput-input MuiInputBase-input"
                  role="button"
                  tabindex="0"
                >
                  <a
                    href="/login"
                  >
                    Login
                  </a>
                </div>
                <input
                  aria-hidden="true"
                  class="c16 MuiSelect-nativeInput"
                  tabindex="-1"
                  value="1"
                />
                <svg
                  aria-hidden="true"
                  class="c17 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium c18 MuiSelect-icon MuiSelect-iconOutlined"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="c19 c20 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="c21"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </header>
        <div
          class="c22"
        />
      </div>
      <div
        data-testid="SuspenseTestId"
      />
    </div>
  </body>,
  "container": .c0 {
  background-color: #ffffff;
  color: rgba(0,0,0,0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

.c6 {
  margin: 0;
  color: #080808;
  font-size: 18px;
  font-family: Source Sans Pro;
  font-weight: 600;
  line-height: 1.167;
}

.c7 {
  margin: 0;
  color: #080808;
  font-size: 16px;
  font-family: Source Sans Pro;
  font-weight: 600;
  line-height: 1.235;
  color: #686868;
}

.c17 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: 1.5rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  box-sizing: border-box;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: fixed;
  z-index: 1100;
  top: 0;
  left: auto;
  right: 0;
  background-color: #00758e;
  color: #fff;
  background-color: #142A35;
}

.c9 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.4375em;
  color: rgba(0,0,0,0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9.Mui-disabled {
  color: rgba(0,0,0,0.38);
  cursor: default;
}

.c14 {
  font: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
}

.c14::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c14::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c14:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c14::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c14:focus {
  outline: 0;
}

.c14:invalid {
  box-shadow: none;
}

.c14::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false] + .MuiInputBase-formControl .c13::-webkit-input-placeholder {
  opacity: 0 !important;
}

label[data-shrink=false] + .MuiInputBase-formControl .c13::-moz-placeholder {
  opacity: 0 !important;
}

label[data-shrink=false] + .MuiInputBase-formControl .c13:-ms-input-placeholder {
  opacity: 0 !important;
}

label[data-shrink=false] + .MuiInputBase-formControl .c13::-ms-input-placeholder {
  opacity: 0 !important;
}

label[data-shrink=false] + .MuiInputBase-formControl .c13:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false] + .MuiInputBase-formControl .c13:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false] + .MuiInputBase-formControl .c13:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false] + .MuiInputBase-formControl .c13:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.c14.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0,0,0,0.38);
}

.c14:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.c19 {
  text-align: left;
  position: absolute;
  bottom: 0;
  right: 0;
  top: -5px;
  left: 0;
  margin: 0;
  padding: 0 8px;
  pointer-events: none;
  border-radius: inherit;
  border-style: solid;
  border-width: 1px;
  overflow: hidden;
  min-width: 0%;
}

.c21 {
  float: unset;
  overflow: hidden;
  padding: 0;
  line-height: 11px;
  -webkit-transition: width 150ms cubic-bezier(0.0,0,0.2,1) 0ms;
  transition: width 150ms cubic-bezier(0.0,0,0.2,1) 0ms;
}

.c10 {
  position: relative;
  border-radius: 4px;
}

.c10:hover .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0,0,0,0.87);
}

.c10.Mui-focused .MuiOutlinedInput-notchedOutline {
  border-color: #00758e;
  border-width: 2px;
}

.c10.Mui-error .MuiOutlinedInput-notchedOutline {
  border-color: #d32f2f;
}

.c10.Mui-disabled .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0,0,0,0.26);
}

.c20 {
  border-color: rgba(0,0,0,0.23);
}

.c15 {
  padding: 16.5px 14px;
}

.c15:-webkit-autofill {
  border-radius: inherit;
}

.c12 {
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-radius: 4px;
  cursor: pointer;
}

.c12:focus {
  border-radius: 4px;
}

.c12::-ms-expand {
  display: none;
}

.c12.Mui-disabled {
  cursor: default;
}

.c12[multiple] {
  height: auto;
}

.c12:not([multiple]) option,
.c12:not([multiple]) optgroup {
  background-color: #ffffff;
}

.c12.c12.c12 {
  padding-right: 32px;
}

.c12.MuiSelect-select {
  height: auto;
  min-height: 1.4375em;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c18 {
  position: absolute;
  right: 7px;
  top: calc(50% - .5em);
  pointer-events: none;
  color: rgba(0,0,0,0.54);
}

.c18.Mui-disabled {
  color: rgba(0,0,0,0.26);
}

.c16 {
  bottom: 0;
  left: 0;
  position: absolute;
  opacity: 0;
  pointer-events: none;
  width: 100%;
  box-sizing: border-box;
}

.c2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 16px;
  padding-right: 16px;
  min-height: 56px;
}

.c3 {
  padding: 0px;
}

.c4 {
  margin-top: 5px;
  height: 45px;
  width: 45px;
}

.c8 {
  margin-left: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c11 {
  width: 90px !important;
  height: 60px !important;
  margin: 0 24px;
  color: #ffffff;
  margin-left: 24px;
  position: relative;
}

.c11 .MuiSelect-select {
  padding: 0 13px 0 0 !important;
}

.c11 fieldset {
  border: none;
}

.c11 .MuiSelect-icon {
  fill: #ffffff;
  right: 0;
  width: 20px;
  height: 20px;
}

.c22 {
  margin: 64px 0 0 0;
}

@media print {

}

@media (hover:none) {

}

@media print {
  .c1 {
    position: absolute;
  }
}

@media (hover:none) {
  .c10:hover .MuiOutlinedInput-notchedOutline {
    border-color: rgba(0,0,0,0.23);
  }
}

@media (min-width:660px) {
  .c2 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:0px) and (orientation:landscape) {
  .c2 {
    min-height: 48px;
  }
}

@media (min-width:660px) {
  .c2 {
    min-height: 64px;
  }
}

<div>
    <div
      style="display: none;"
    >
      <header
        class="c0 MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 c1 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed"
      >
        <div
          class="c2 MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular c3"
        >
          <a
            href="/"
            style="padding: 5px 10px 5px 0px;"
          >
            <svg
              class="c4"
            >
              macmillan_flag.svg
            </svg>
          </a>
          <div
            class="c5"
          >
            <h3
              class="c6 MuiTypography-root MuiTypography-h3"
            />
            <h4
              class="c7 MuiTypography-root MuiTypography-h4"
            />
          </div>
          <div
            class="c8"
          >
            <div
              class="c9 c10 MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiSelect-root-kTniIo c11"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                class="c12 MuiSelect-select MuiSelect-outlined c13 c14 c15 MuiOutlinedInput-input MuiInputBase-input"
                role="button"
                tabindex="0"
              >
                <a
                  href="/login"
                >
                  Login
                </a>
              </div>
              <input
                aria-hidden="true"
                class="c16 MuiSelect-nativeInput"
                tabindex="-1"
                value="1"
              />
              <svg
                aria-hidden="true"
                class="c17 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium c18 MuiSelect-icon MuiSelect-iconOutlined"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="c19 c20 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="c21"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </header>
      <div
        class="c22"
      />
    </div>
    <div
      data-testid="SuspenseTestId"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
